### DISCUSSION TITLE
Is react-router v7 still in beta?

### DISCUSSION THREAD

**Discussion by judocode** [remix-run/react-router#12533] [Q&A]
I have yet to upgrade any Remix v2 to react-router v7 successfully. Doesn't work in a npm monorepo, nor a pnpm monorepo. And browsing the repo, I see tons of issues with v7. Even the codemod listed as "performing the majority of steps 2-8" in the migration guide only does 2 of the 6 steps. It feels like this release was rushed and is not quite ready for production for even the happy path.

I've tried to upgrade several times now over the last few weeks and still am running into known issues, where I cannot even get the app to start, so I'm wondering when we can expect v7 to be stable?

**Comment by sergiodxa** [comment]
v7.0.0 stable was released 3 weeks ago (see https://github.com/remix-run/react-router/releases/tag/react-router%407.0.1)
